addons/mail/models/res_users.py

```diff
@@ -66,6 +76,11 @@ def _inverse_notification_type(self):
         inbox_users.write({"groups_id": [Command.link(inbox_group.id)]})
         (self - inbox_users).write({"groups_id": [Command.unlink(inbox_group.id)]})
 
+    @api.depends('groups_id')
+    def _compute_is_readonly_mail_role_ids(self):
+        for user in self:
+            user.is_readonly_mail_role_ids = not self.env['mail.role'].has_access('write')
```

Comment:
the loop is not necessary in this case, the same value can be written on all records